### PR TITLE
Provide more information about the 'Support' filter to avoid confusion

### LINKED ISSUE #70598: No results appear in project manager asset library if using the Open Asset Library button
### Godot version

v4.0.beta10.official.d0398f62f

### System information

Windows 10, NVIDIA GeForce GTX 750 Ti/PCIe/SSE2

### Issue description

Hi,

when run Godot_v4.0-beta10_win64
it's show dialog question if "Open Asset Library" or "Cancel"
if I click "Open Asset Library" it's not shwing any thing
but if click cancel and click "Open Asset Library" from tab its show demos

preview problem in video
https://drive.google.com/file/d/1hZTgMiJLLk8Cz1B_voX25CdNHKeuCXyH/view?usp=sharing

### Steps to reproduce

https://user-images.githubusercontent.com/5288136/209568828-c8824202-5b9c-4df3-a6a2-e838c3b88ac6.mp4

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by Maran23**
Fixes https://github.com/godotengine/godot/issues/70598
-> May help to avoid confusion as written down in the issue.

| Before | After |
|-|-|
| image | image |
| image | image |

### CODE REVIEW

**Comment by Calinou** [comment]
I would suggest explicitly listing the Support levels that were chosen. For example:

> No results compatible with Godot 4.0 for support levels "Official".
> Check the enabled support levels using the Support button in the top-right corner.

**Comment by Maran23** [comment]
> This is a good start, but I suggest explicitly listing the support levels that were chosen.

Done. I also added this to the normal filter search result, as it can also happen there.
I also updated the screenshots.
Open for feedback. :)

**Comment by pwab** [comment]
Hey @Maran23 thanks for this little but helpful addition. I also was confused about this message and apart from the missing explanation this could be triggered two other things:

1. I did not realize that the support-filter button is in fact ... a button. It is styled like the labels (Sort, Category, Site) without any border or highlight (color, shading) to make it look like a button. Also, all dropdowns in this window have this little triangle on the right side to indicate that the field is a dropdown element. The support-filter button does not have a triangle.
2. The naming 'Support' might also be confusing. When looking at it in this menu, my thought as a first time user is 'This must lead to the asset-lib support page'. But I have no clue what kind of renaming could help here.

Would it be possible to incorporate some minor UI changes next to the textual explanation to improve this dialog even further?

**Comment by Maran23** [comment]
> 1. I did not realize that the support-filter button is in fact ... a button. It is styled like the labels (Sort, Category, Site) without any border or highlight (color, shading) to make it look like a button. Also, all dropdowns in this window have this little triangle on the right side to indicate that the field is a dropdown element. The support-filter button does not have a triangle.

I agree. Unfortunately this is probably a bigger task, since this component (`MenuButton`) is used more often around the engine. So not only this location suffers from this.
image

We may want to open a proposal to improve the look and feel of `MenuButton`s.

> 2. The naming 'Support' might also be confusing. When looking at it in this menu, my thought as a first time user is 'This must lead to the asset-lib support page'. But I have no clue what kind of renaming could help here.

Not sure here too. I also find this name quite confusing. This is more likely some kind of category.

This PR is just a low hanging fruit, trying to improve the current state without much refactoring or more.

File: editor/plugins/asset_library_editor_plugin.cpp [hunk 1]
```diff
@@ -1242,15 +1248,28 @@ void EditorAssetLibrary::_http_request_completed(int p_status, int p_code, const
...
+					}
+				}
+				if (support_list.is_empty()) {
+					support_list = "-";
+				}
+
 				if (!filter->get_text().is_empty()) {
 					library_info->set_text(
-							vformat(TTR("No results for \"%s\"."), filter->get_text()));
+							vformat(TTR("No results for \"%s\" for support levels: %s."), filter->get_text(), support_list));
```
Comment (akien-mga) [inline editor/plugins/asset_library_editor_plugin.cpp:1266]:
```suggestion
							vformat(TTR("No results for \"%s\" for support level(s): %s."), filter->get_text(), support_list));
```

File: editor/plugins/asset_library_editor_plugin.cpp [hunk 2]
```diff
@@ -1242,15 +1248,28 @@ void EditorAssetLibrary::_http_request_completed(int p_status, int p_code, const
...
 				} else {
 					// No results, even though the user didn't search for anything specific.
 					// This is typically because the version number changed recently
 					// and no assets compatible with the new version have been published yet.
 					library_info->set_text(
-							vformat(TTR("No results compatible with %s %s."), String(VERSION_SHORT_NAME).capitalize(), String(VERSION_BRANCH)));
+							vformat(TTR("No results compatible with %s %s for support levels: %s.\nCheck the enabled support levels using the 'Support' button in the top-right corner."), String(VERSION_SHORT_NAME).capitalize(), String(VERSION_BRANCH), support_list));
```
Comment (akien-mga) [inline editor/plugins/asset_library_editor_plugin.cpp:1272]:
```suggestion
							vformat(TTR("No results compatible with %s %s for support level(s): %s.\nCheck the enabled support levels using the 'Support' button in the top-right corner."), String(VERSION_SHORT_NAME).capitalize(), String(VERSION_BRANCH), support_list));
```

File: editor/plugins/asset_library_editor_plugin.cpp [hunk 3]
```diff
@@ -1242,15 +1248,28 @@ void EditorAssetLibrary::_http_request_completed(int p_status, int p_code, const
 			library_vb->add_child(asset_bottom_page);
 
 			if (result.is_empty()) {
+				String support_list;
+				for (int i = 0; i < SUPPORT_MAX; i++) {
+					if (support->get_popup()->is_item_checked(i)) {
+						if (!support_list.is_empty()) {
+							support_list += ", ";
+						}
+						support_list += TTR(support_ttr_key[i]);
```
Comment (akien-mga) [inline editor/plugins/asset_library_editor_plugin.cpp:1257]: This won't work to extract the translation strings, `TTR()` needs to be used with a literal.
You should use `TTRGET()` here, and `TTRC()` in the Array.

File: editor/plugins/asset_library_editor_plugin.cpp [hunk 4]
```diff
@@ -1510,9 +1529,9 @@ EditorAssetLibrary::EditorAssetLibrary(bool p_templates_only) {
 	search_hb2->add_child(support);
 	support->set_text(TTR("Support"));
 	support->get_popup()->set_hide_on_checkable_item_selection(false);
-	support->get_popup()->add_check_item(TTR("Official"), SUPPORT_OFFICIAL);
-	support->get_popup()->add_check_item(TTR("Community"), SUPPORT_COMMUNITY);
-	support->get_popup()->add_check_item(TTR("Testing"), SUPPORT_TESTING);
+	support->get_popup()->add_check_item(TTR(support_ttr_key[SUPPORT_OFFICIAL]), SUPPORT_OFFICIAL);
+	support->get_popup()->add_check_item(TTR(support_ttr_key[SUPPORT_COMMUNITY]), SUPPORT_COMMUNITY);
+	support->get_popup()->add_check_item(TTR(support_ttr_key[SUPPORT_TESTING]), SUPPORT_TESTING);
```
Comment (akien-mga) [inline editor/plugins/asset_library_editor_plugin.cpp:1534]: Same comment as above, this won't work, should use `TTRGET()`.

File: editor/plugins/asset_library_editor_plugin.h [hunk 5]
```diff
@@ -234,6 +234,7 @@ class EditorAssetLibrary : public PanelContainer {
 	static const char *sort_key[SORT_MAX];
 	static const char *sort_text[SORT_MAX];
 	static const char *support_key[SUPPORT_MAX];
+	static const char *support_ttr_key[SUPPORT_MAX];
```
Comment (akien-mga) [inline editor/plugins/asset_library_editor_plugin.h:237]: Could be `support_text` to be consistent with `sort_key`/`sort_text`.

**Review by akien-mga — CHANGES_REQUESTED** [review]

File: editor/plugins/asset_library_editor_plugin.cpp (hunk 3 above)
Comment (Maran23) [reply editor/plugins/asset_library_editor_plugin.cpp]: Ahh, I see, thanks! 
Done. (All your comments)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**